doc/designs/quic-design/quic-overview.md

```diff
@@ -0,0 +1,128 @@
+QUIC Design Overview
+====================
+
+The QUIC implementation in OpenSSL is roughly described by the following
+picture.
+
+![alt_text](images/quic-overview.svg "QUIC Implementation Building Blocks")
+
+SSL API
+-------
+
+The application facing public API of the OpenSSL library.
+
+Stream Send and Read Buffers
+----------------------------
+
+Buffers for stream data to be sent or received from the peer over the
+QUIC protocol. These are necessary to support existing semantics of the
+SSL_read and SSL_write functions.
+
+They will be bypassed with a single-copy API for read and write (_not
+for MVP_).
+
+Connection State Machine
+------------------------
+
+A state machine handling the state for a QUIC connection.
+
+Connection ID Cache
+-------------------
+
+A table matching Connection IDs with Connection objects represented
+via SSL objects.
+
+_In MVP there is basically 1-to-1 matching of Connection IDs to Connection
+objects. This will be changed with Path migration support and QUIC
+server support._
+
+Timer And Event Queue
+---------------------
+
+Queue of events that need to be handled asynchronously or at a later
+time.
+
+TLS Handshake Record Layer
+--------------------------
+
+A module that uses the Record Layer API to implement the inner TLS-1.3
+protocol handshake. It produces and parses the QUIC CRYPTO frames.
+
+TX Packetizer
+-------------
+
+This module creates frames from the application data obtained from
+the application. It also receives CRYPTO frames from the TLS Handshake
+Record Layer and ACK frames from the ACK Handling And Loss Detector
+subsystem.
+
+RX Frame Handler
+----------------
+
+Decrypted packets are split into frames here and the frames are forwarded
+either as data or as events to the subsequent modules based on the frame
+type. Flow Controller And Statistics Collector is consulted for decisions
+and to record the statistics of the received stream data.
+
+Flow Controller And Statistics Collector
+----------------------------------------
+
+This module collects various statistics about send and received
+stream data. It is also consulted by the TX Packetizer and RX Frame
+Handler for flow control decisions.
+
+QUIC Write Record Layer
+-----------------------
+
+Encryption of packets according to the given encryption level and with
+the appropriate negotiated algorithm happens here.
+
+Resulting packets are sent through the Datagram BIO interface to the
+network.
+
+QUIC Read Record Layer
+----------------------
+
+Decryption of packets according to the given encryption level and with
+the appropriate negotiated algorithm happens here.
+
+Packets are received from the network through the Datagram BIO interface.
+
+Congestion Controller
+---------------------
+
+This is a pluggable API that provides calls to collect data relevant
```

Comment:
s/collect/receive/ ?  I feel that _collect_ is the wrong word since that's not what's going on.

Reply:
I've used record.